src/CleverDict.jl

```diff
@@ -0,0 +1,177 @@
+import OrderedCollections
+
+"""
+    CleverDict{K, V}
```

Comment:
Consider adding a few unit tests for this data structure.

Reply:
It should probably be moved to MOI as it might be used by LQOI solvers (and maybe more).
In MOI, we currently use a `Vector{Union{Nothing, V}}` instead when we are in the use case of `CleverDict`. Not sure which one is better.